presto-product-tests/src/main/resources/sql-tests/testcases/window_functions/windowSpecificationBcpAA.sql

```diff
@@ -0,0 +1,8 @@
+-- database: presto; groups: window;
+select
+suppkey, orderkey, partkey,
+nth_value(quantity, 4) over (partition by partkey order by orderkey rows between UNBOUNDED preceding and CURRENT ROW) nth_value_quantity_B,
+sum(quantity) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_quantity_A,
+sum(discount) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_discount_A
```

Comment:
The CTE case is straightforward for a single CTE: The WindowNodes in the CTE show up basically as they do in a normal query.

If you have 2 CTEs and join them, we do not merge functions from WindowNodes with identical specifications across the left and right sources of the join. Without digging deeper, this seems likely to be the correct behavior. The reason this happens is that the symbol for the result of the function is different. If it were to be the same we'd merge. As I mentioned, I'm skeptical about this absent some quality time with some smart people around a whiteboard.

If we're OK counting on the symbols on the left and right sides of the join being different, we can leave MIW as-is. If not, we need to create new context for each side of the join.

Thoughts? @kokosing, @cawallin, @maciejgrzybek